### DISCUSSION TITLE
Introduce cache for unpaged lists?

### DISCUSSION THREAD

**Discussion by dvv** [kanidm/kanidm#2669] [General]
Hi!

I wonder if we could have (already have?) a native cache for `/v1/(person|group)` handlers?
On my setup they dump circa 200/100kB and as such are IMO certainly open to kinda DDOS.
If it's not relevant to Kani per se, what could be a hook to invalidate external cache?

TIA

**Comment by yaleman** [comment]
There's a lot of caching internally for entries and indexes, but not for the actual HTTP responses. The work to disable anonymous queries would avoid a lot of the bandwidth-related risk.

For caching, we'd need to implement [ETag headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag) and support HEAD requests for the endpoints, which is not a _bad_ idea, but wouldn't be a high priority for at least a release or two.

**Reply by dvv** [reply]
Right.
I've no anonymous queries AFAIKS but there can be malicious well-authenticated persons, or dumb/buggy shell scripts etc.
I since realized that caching should be ACP-wise (in sense of, e.g., whether to expose legalname and to whom) and that sounds almost a dead-end, at least for near future.
Tnx!

**Reply by yaleman** [reply]
Test it, I'd be surprised if it was a serious issue unless you're intentionally constraining the system resources.

**Reply by dvv** [reply]
It's come from a habit to _a priori_ assume the corner cases, hence the discussion.
